tm/src/main/java/io/seata/tm/api/transaction/TransactionHookManager.java

```diff
@@ -61,6 +88,33 @@ public static void registerHook(TransactionHook transactionHook) {
         LOCAL_HOOKS.get().add(transactionHook);
     }
 
+    /**
+     * trigger hooks
+     *
+     * @param xid
+     * @param trigger
+     */
+    public static void triggerHooks(String xid, Consumer<List<TransactionHook>> trigger) {
+        List<TransactionHook> hooks = getHooks();
+        if (hooks == null || hooks.isEmpty()) {
+            return;
+        }
+
+        boolean inGlobalTransaction = RootContext.inGlobalTransaction();
+
+        try {
+            if (!inGlobalTransaction) {
```

Comment:
why bind？ Before  begin process，get xid is null.

Reply:
some hook can't get xid.